service/debugger/locations.go

```diff
@@ -303,8 +304,33 @@ func (loc *NormalLocationSpec) FileMatch(path string) bool {
 	return partialPathMatch(loc.Base, path)
 }
 
-func tryMatchRelativePathByProc(expr, debugname, file string) bool {
-	return len(expr) > 0 && expr[0] == '.' && file == path.Join(path.Dir(debugname), expr)
+func (loc *NormalLocationSpec) tryMatchFullPath(d *Debugger, file string) bool {
+	locBaseAbsPath := loc.Base
+	if !filepath.IsAbs(locBaseAbsPath) && len(d.processArgs) >= 1 {
+		locBaseAbsPath = path.Join(path.Dir(d.processArgs[0]), loc.Base)
+	}
+	if !filepath.IsAbs(locBaseAbsPath) {
+		return false
+	}
+
+	curFile := file
+	for {
```

Comment:
Couldn't we replace this loop with https://godoc.org/path/filepath#EvalSymlinks ?

Reply:
Ex: ...-> a.go -> b.go -> c.go -> d.go, and we build b.go into b.  
The target of b contian information of b.go.  
The key point is I think set breakpoint on b.go/c.go/d.go for b is ok, but a.go not (Maybe I'm wrong, what's your opinion?).  
So if we use EvalSymlinks(a.go/b.go/c.go) == d.go, we can't distinguish between a.go and c.go.